### PR TITLE
pylint and black

### PR DESCRIPTION
**Pull request by linuxrider**

### CODE REVIEW

File: ec_tools/helper.py [hunk 1]
```diff
@@ -64,6 +69,7 @@ def determine_scan_rate(t, x):
 
     return np.abs(np.diff(x) / np.diff(t)).mean()
 
+
 def detect_step(t, x):
     """Return index of step in given t and x arrays.
     Index is the where the changed value of t located.
```
Comment (DunklesArchipel) [inline ec_tools/helper.py:75]: This is a somewhat incomplete sentence.

File: ec_tools/semi_integration.py [hunk 2]
```diff
@@ -3,19 +3,20 @@
...
 - evaluate if it is possible have varying Δx i.e. passing x array
 """
 
 import numpy as np
 
-def prepare_kernel(q, Δx, N, c1, c2):
+
+def prepare_kernel(q, delta_x, N, c1, c2):
     r"""
     Setup the integration kernel with the order q, the x interval Δx, the length of the array N,
```
Comment (DunklesArchipel) [inline ec_tools/semi_integration.py:15]:
```suggestion
    Setup the integration kernel with the order q, the x interval delat_x, the length of the array N,
```

File: ec_tools/semi_integration.py [hunk 3]
```diff
@@ -25,19 +26,20 @@ def prepare_kernel(q, Δx, N, c1, c2):
...
+        t_j = tau0 * np.exp(-j / (q * c2))
+        w1[i] = t_j / (delta_x + t_j)
         w2[i] = a_j * (1 - w1[i])
     return s, w1, w2
 
-def semi_integration(y, q=-0.5, Δx=1, c1=8, c2=2):
+
+def semi_integration(y, q=-0.5, delta_x=1, c1=8, c2=2):
     """
     Return the semiintegral R of order q for y with the x interval Δx and the filter constants
```
Comment (DunklesArchipel) [inline ec_tools/semi_integration.py:40]:
```suggestion
    Return the semiintegral R of order q for y with the x interval delta_x and the filter constants
```

File: ec_tools/semi_integration.py [hunk 4]
```diff
@@ -25,19 +26,20 @@ def prepare_kernel(q, Δx, N, c1, c2):
...
     c1 and c2.
-    
+
     Semi-integrating two times with order q = -0.5 should give the same result as integrating once.
     The relative error should not exceed 0.25 percent for 1000 and 0.5 percent per 10000 integration steps.
     TODO:
```
Comment (DunklesArchipel) [inline ec_tools/semi_integration.py:45]: Maybe you should open an issue for the Todo and add the issue number
```suggestion
    TODO:: see #X
```

File: ec_tools/helper.py [hunk 5]
```diff
@@ -12,7 +12,7 @@ def find_x0_values(x, y, mode="all"):
 
     TODO:
```
Comment (DunklesArchipel) [inline ec_tools/helper.py:13]: Maybe you should open an issue for the Todo and add the issue number
```suggestion
    TODO:: see #X
```

**Review by DunklesArchipel — APPROVED** [review]
looks good. Requested changes address in parts already merged content.

File: ec_tools/helper.py [hunk 6]
```diff
@@ -64,6 +69,7 @@ def determine_scan_rate(t, x):
 
     return np.abs(np.diff(x) / np.diff(t)).mean()
 
+
 def detect_step(t, x):
     """Return index of step in given t and x arrays.
```
Comment (linuxrider) [inline ec_tools/helper.py:74]:
```suggestion
    """Returns the index of the step in given t and x arrays.
```

File: ec_tools/helper.py (hunk 5 above)
Comment (linuxrider) [reply ec_tools/helper.py:13]:
```suggestion
    TODO:: see #5
```

File: ec_tools/semi_integration.py (hunk 4 above)
Comment (linuxrider) [reply ec_tools/semi_integration.py:45]:
```suggestion
    TODO:: see #6
```

**Pull request merged**